### PR TITLE
Behat suite for PS Command using stubbed infrastructure

### PR DESCRIPTION
**Pull request by ciaranmcnulty**
Two suites:
- Application - uses Symfony application tester and stubs out the shell commands with fakes
- Smoke - executes CLI command directly and assumes host has lots of stuff already installed

Smoke suite should probably run the `docker:install` command eventually, and needs to do more setup/teardown when further scenarios are added.

### CODE REVIEW

File: features/bootstrap/SmokeContext.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    {
+        $this->fs = new \Symfony\Component\Filesystem\Filesystem();
+    }
+
+    /**
+     * @Given I have a Docker Compose file that contains one container
+     */
+    public function iHaveADockerComposeFileThatContainsOneContainer()
+    {
+        $path = __DIR__ . '/../../../tmp/';
```
Comment (sroze) [inline features/bootstrap/SmokeContext.php]: You should remove a level to be at project root

File: features/bootstrap/SmokeContext.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        $this->lastOutput = shell_exec(__DIR__ . '/../../bin/dock-cli '. escapeshellarg($command));
+    }
+
+
+    /**
+     * @Then I should see that this container has a status of :status
+     */
+    public function iShouldSeeThatThisContainerHasAStatusOf($status)
+    {
+        if (!preg_match('/|\s+running\s+|$/', $this->lastOutput)) {
```
Comment (ciaranmcnulty) [inline features/bootstrap/SmokeContext.php]: This should use `$status` not hard-code `running`

File: features/bootstrap/SmokeContext.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+        $this->fs->dumpFile('docker-compose.yml', $config);
+    }
+
+    /**
+     * @Given this container is running
+     */
+    public function thisContainerIsRunning()
+    {
+        shell_exec(__DIR__ . '/../../bin/dock-cli up');
```
Comment (sroze) [inline features/bootstrap/SmokeContext.php]: Might be good to move this `__DIR__ . '/../../bin/dock-cli up'` used in many places at least in a private method.

File: src/Cli/Helper/ContainerList.php [hunk 4]
```diff
@@ -48,7 +48,7 @@ public function render(array $containers)
     }
 
     /**
-     * @param Container $container
+     * @param \Dock\Containers\Container $container
```
Comment (sroze) [inline src/Cli/Helper/ContainerList.php]: Why this FQCN ?

File: src/Cli/PsCommand.php [hunk 5]
```diff
@@ -11,14 +11,14 @@
...
+     * @var ConfiguredContainers
      */
     private $inspector;
 
     /**
-     * @param Inspector $inspector
+     * @param ConfiguredContainers $inspector
      */
-    public function __construct(Inspector $inspector)
+    public function __construct(ConfiguredContainers $inspector)
```
Comment (sroze) [inline src/Cli/PsCommand.php]: So you should change the variable name

File: src/Containers/ConfiguredContainers.php [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+     * @return array
+     */
+    private function getRunningContainerIds()
+    {
+        return $this->configuredContainerIds->findAll();
+    }
+
+    /**
+     * @param string $containerId
+     * @return array
```
Comment (sroze) [inline src/Containers/ConfiguredContainers.php]: `Container[]` ?

File: src/Cli/Helper/ContainerList.php (hunk 4 above)
Comment (ciaranmcnulty) [inline src/Cli/Helper/ContainerList.php]: IDE being helpful when we moved it

**Comment by sroze** [comment]
Awesome ! :+1:

**Pull request merged**